src/main/java/app/attestation/server/AttestationServer.java

```diff
@@ -1146,6 +1142,104 @@ public void handlePost(final HttpExchange exchange) throws IOException, SQLiteEx
         }
     }
 
+    private static class AttestationHistoryHandler extends PostHandler {
+        @Override
+        public void handlePost(final HttpExchange exchange) throws IOException, SQLiteException {
+            String fingerprint;
+            int directive;
+            // 0 for backwards, 1 for forward
+            long offsetId;
+            final Account account = verifySession(exchange, false, null);
+            if (account == null) {
+                return;
+            }
+            try (final JsonReader reader = Json.createReader(exchange.getRequestBody())) {
+                final JsonObject object = reader.readObject();
+                fingerprint = object.getString("fingerprint");
+                directive = Int.valueOf(object.getString("directive"));
```

Comment:
If this is a number in the JSON request, you should obtain it with `getInt` instead of trying to obtain it as a string and then converting to an integer.